e2e-tests/gef/cypress/integration/unissued-facilities-change-all-AIN.spec.js

```diff
@@ -0,0 +1,364 @@
+      aboutFacilityUnissued.coverStartDateMonth().type(twoMonthsMonth);
+      aboutFacilityUnissued.coverStartDateYear().type(twoMonthsYear);
+
+      aboutFacilityUnissued.coverEndDateDay().type(threeMonthsOneDayDay);
+      aboutFacilityUnissued.coverEndDateMonth().type(threeMonthsOneDayMonth);
+      aboutFacilityUnissued.coverEndDateYear().type(threeMonthsOneDayYear);
+      aboutFacilityUnissued.continueButton().click();
+
+      unissuedFacilityTable.successBanner().contains(`${unissuedFacilitiesArray[0].name} is updated`);
+      unissuedFacilityTable.rows().should('have.length', unissuedFacilitiesArray.length - 1);
+      unissuedFacilityTable.continueButton().should('not.exist');
+
+      unissuedFacilityTable.updateIndividualFacilityButton(0).click();
+      aboutFacilityUnissued.issueDateDay().type(oneMonthDay);
+      aboutFacilityUnissued.issueDateMonth().type(oneMonthMonth);
+      aboutFacilityUnissued.issueDateYear().type(oneMonthYear);
+
+      aboutFacilityUnissued.shouldCoverStartOnSubmissionNo().click();
+      aboutFacilityUnissued.coverStartDateDay().type(twoMonthsDay);
+      aboutFacilityUnissued.coverStartDateMonth().type(twoMonthsMonth);
+      aboutFacilityUnissued.coverStartDateYear().type(twoMonthsYear);
+      aboutFacilityUnissued.coverEndDateDay().type(threeMonthsOneDayDay);
+      aboutFacilityUnissued.coverEndDateMonth().type(threeMonthsOneDayMonth);
+      aboutFacilityUnissued.coverEndDateYear().type(threeMonthsOneDayYear);
+      aboutFacilityUnissued.continueButton().click();
+
+      unissuedFacilityTable.successBanner().contains(`${unissuedFacilitiesArray[1].name} is updated`);
+      unissuedFacilityTable.rows().should('have.length', unissuedFacilitiesArray.length - 2);
+      unissuedFacilityTable.continueButton().should('not.exist');
+
+      unissuedFacilityTable.updateIndividualFacilityButton(0).click();
+      aboutFacilityUnissued.issueDateDay().type(oneMonthDay);
+      aboutFacilityUnissued.issueDateMonth().type(oneMonthMonth);
+      aboutFacilityUnissued.issueDateYear().type(oneMonthYear);
+
+      aboutFacilityUnissued.shouldCoverStartOnSubmissionNo().click();
+      aboutFacilityUnissued.coverStartDateDay().type(twoMonthsDay);
+      aboutFacilityUnissued.coverStartDateMonth().type(twoMonthsMonth);
+      aboutFacilityUnissued.coverStartDateYear().type(twoMonthsYear);
+
+      aboutFacilityUnissued.coverEndDateDay().type(threeMonthsOneDayDay);
+      aboutFacilityUnissued.coverEndDateMonth().type(threeMonthsOneDayMonth);
+      aboutFacilityUnissued.coverEndDateYear().type(threeMonthsOneDayYear);
+      aboutFacilityUnissued.continueButton().click();
+
+      unissuedFacilityTable.rows().should('have.length', 0);
+      unissuedFacilityTable.allUnissuedUpdatedSuccess().contains('Facility Stages are now updated');
+      unissuedFacilityTable.continueButton().should('exist');
+      unissuedFacilityTable.continueButton().click();
+    });
```

Comment:
There's a lot going on in this spec and the previous one (excellent testing), but bit hard to understand what's going on from reading. Could we add some comments? ie.


```js
// submit invalid xyz
aboutFacilityUnissued.coverEndDateYear().type(threeMonthsOneDayYear);
aboutFacilityUnissued.continueButton().click();

// should get taken to X and error messages displayed
cy.url().should('eq', ....
```

Reply:
done